### PR TITLE
MM-1773a Add wildcard searching for postgresql

### PR DESCRIPTION
**Pull request by nickago**

### CODE REVIEW

**Comment by coreyhulen** [comment]
unit tests?

File: store/sql_post_store.go [hunk 1]
```diff
@@ -386,6 +388,12 @@ func (s SqlPostStore) Search(teamId string, userId string, terms string, isHasht
 		var posts []*model.Post
 
 		if utils.Cfg.SqlSettings.DriverName == "postgres" {
+
+			// Parse text for wildcards
+			if wildcard, err := regexp.Compile("\\*($| )"); err == nil {
+				terms = wildcard.ReplaceAllLiteralString(terms, ":* ")
```
Comment (coreyhulen) [inline store/sql_post_store.go]: does this work when it's the end of the line?  my search term is "corey*" vs "corey* "

Comment (coreyhulen) [inline store/sql_post_store.go]: nvm

**Comment by nickago** [comment]
Right, of course

**Comment by nickago** [comment]
@coreyhulen Ready for re-review, unit tests have been added

**Comment by crspeller** [comment]
+1

**Comment by coreyhulen** [comment]
+1

**Comment by crspeller** [comment]
@nickago This needs rebase

**Comment by nickago** [comment]
Rebase complete

**Comment by crspeller** [comment]
@nickago This now fails to compile. Make sure you always test after a rebase. git is not perfect.

**Comment by nickago** [comment]
Of course, changes are made

**Pull request merged**